### PR TITLE
Add UI Improvements to Server Configurations

### PR DESCRIPTION
**Pull request by vihanga-liyanage**
## Purpose
Added the following UI improvements.
- Add summary views
- Add grid layout
- Add toggles instead of regular checkboxes.

### CODE REVIEW

File: apps/admin-portal/src/components/server-configurations/account-recovery.tsx [hunk 1]
```diff
@@ -101,46 +100,123 @@ export const AccountRecovery: FunctionComponent<AccountRecoveryProps> = (props:
...
+				handleConfirmationModalClose();
+				hideFormEditView(ACCOUNT_RECOVERY_FORM_IDENTIFIER);
+			})
+			.catch((error) => {
+				// Axios throws a generic `Network Error` for 401 status.
+				// As a temporary solution, a check to see if a response is available has been used.
+				if (!error.response || error.response.status === 401) {
+					dispatch(addAlert(errorMessage));
+				} else if (error.response && error.response.data && error.response.data.detail) {
+
```
Comment (thivi) [inline apps/admin-portal/src/components/server-configurations/account-recovery.tsx]: Shall we remove this line?

File: apps/admin-portal/src/components/server-configurations/user-self-registration.tsx [hunk 2]
```diff
@@ -268,127 +278,208 @@ export const UserSelfRegistration: FunctionComponent<UserSelfRegistrationProps>
...
+							label: t("devPortal:components.serverConfigs.selfRegistration.form." +
+								"enable.label"),
 							value: ServerConfigurationsConstants.SELF_REGISTRATION_ENABLE
-						},
+						}
+					] }
+					value={ selfSignUpConfigs.enable }
+					listen={
+						(values) => {
+							const value = values.get(ServerConfigurationsConstants.SELF_REGISTRATION_ENABLE).length > 0
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/server-configurations/user-self-registration.tsx]: Does this line exceeds 120 character limit? Is so shall we break it into two lines?

File: apps/admin-portal/src/components/server-configurations/user-self-registration.tsx [hunk 3]
```diff
@@ -268,127 +278,208 @@ export const UserSelfRegistration: FunctionComponent<UserSelfRegistrationProps>
...
+						{
+							label: t("devPortal:components.serverConfigs.selfRegistration.form." +
+								"enableReCaptcha.label"),
+							value: ServerConfigurationsConstants.RE_CAPTCHA
+						}
+					] }
+					value={ selfSignUpConfigs.reCaptcha }
+					listen={
+						(values) => {
+							const value = values.get(ServerConfigurationsConstants.RE_CAPTCHA).length > 0
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/server-configurations/user-self-registration.tsx]: Shall we add a null check as well?

File: apps/admin-portal/src/components/server-configurations/user-self-registration.tsx [hunk 4]
```diff
@@ -268,127 +278,208 @@ export const UserSelfRegistration: FunctionComponent<UserSelfRegistrationProps>
...
-						size="small"
-						type="button"
-						value={ t("common:cancel").toString() }
-					/>
-				</Form.Group>
+			</Forms>
+		</EditSection>
+	);
 
+	const showUserSelfRegistrationView = editingForm[USER_SELF_REGISTRATION_FORM_IDENTIFIER] ? (
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/server-configurations/user-self-registration.tsx]:
```suggestion
	const showUserSelfRegistrationView = editingForm[USER_SELF_REGISTRATION_FORM_IDENTIFIER] && (
```

File: apps/admin-portal/src/components/server-configurations/user-self-registration.tsx [hunk 5]
```diff
@@ -268,127 +278,208 @@ export const UserSelfRegistration: FunctionComponent<UserSelfRegistrationProps>
...
+									type="button"
+									value={ t("common:cancel").toString() }
+								/>
+							</Form.Group>
+						</Grid.Column>
+					</Grid.Row>
+				</Grid>
 			</Forms>
 		</EditSection>
 	) : null;
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/server-configurations/user-self-registration.tsx]:
```suggestion
	);
```

File: apps/admin-portal/src/components/server-configurations/user-self-registration.tsx (hunk 2 above)
Comment (vihanga-liyanage) [reply apps/admin-portal/src/components/server-configurations/user-self-registration.tsx]: No. it's at the right length.

File: apps/admin-portal/src/components/server-configurations/account-recovery.tsx (hunk 1 above)
Comment (vihanga-liyanage) [reply apps/admin-portal/src/components/server-configurations/account-recovery.tsx]: Done

**Pull request merged**